torch/_inductor/fx_passes/quantization.py

```diff
@@ -653,10 +635,13 @@ def __init__(self, op_name: str, scalars_attr=None, algorithm_attr=None):
                 dtype=original_pattern_output_dtype,
             ),
             UnaryAttr("hardswish", [], ""): generate_pattern_with_output_quant(
-                generate_pattern_with_unary(
-                    get_dequantize_qconv_pt2e_pattern(2), aten.hardswish.default
+                _unary_fusion_pattern(
+                    _hardswish_fusion,
+                    get_dequantize_qconv_pt2e_pattern(1 if is_bf16 else 2),
+                    2,
+                    is_bf16,
                 ),
-                dtype=original_pattern_output_dtype,
+                dtype=torch.float32,
```

Comment:
In what case, is this `dtype` a type other than float32? Or is it always float32?

Reply:
I think here the `dtype` parameter for `generate_pattern_with_output_quant` is used to decide whether `to_fp32` node existing before the quant node. However, in this case, the pattern is `conv->to_fp32->hardswish->quant`, there is no `to_fp32` before the `quant` node since we already to_fp32 before post op.

Reply:
ok, thanks. The name `dtype` doesn't seem specifically named for this purpose. Better we name it like `has_to_fp32_before_quant`?

Reply:
@zhuhaozhe could you help on this?

Reply:
Sure, updated in https://github.com/pytorch/pytorch/pull/122374